### PR TITLE
4726 add rnr reference

### LINKED ISSUE #4726: Add Reference to R&R
## Is your feature request related to a problem? Please describe 👀
Request from client to have a reference on the R&R detail view, such as the one on internal orders:

image

This would typically go into the toolbar area. We've had space issues using the forms on tablets though, so I would add to the RH more info panel.

which means creating a more info panel. You could implement #4727 first 😉 

When creating the internal order from the R&R, copy the reference over.

## Describe the solution you'd like 🎁
Add the reference field and pass it through to the internal order.

### Moneyworks Jobcode 🧰

`DFAT:TDEV`

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #4726

# 👩🏻‍💻 What does this PR do?

Adds comment and reference to R&R form side panel:
Screenshot 2024-10-16 at 1 07 31 PM

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Create/open R&R form
- [ ] Open side panel
- [ ] Add comment & reference
- [ ] Close/open the side panel / refresh - your inputs are persisted
- [ ] Finalise the R&R form
- [ ] You can't edit the fields anymore
- [ ] Reference has been applied to the generated Internal Order

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [x] **These areas should be updated or checked**: 
  1. Side panel docs/explanation
  2.

### CODE REVIEW

File: client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx [hunk 1]
```diff
@@ -53,6 +63,31 @@ export const SidePanel = ({ data }: { data: RnRFormFragment }) => {
             <PanelLabel>{t('label.created')}</PanelLabel>
             <PanelField>{localisedDate(data.createdDatetime)}</PanelField>
           </PanelRow>
+          <PanelRow>
+            <PanelLabel flex={0.4}>{t('heading.reference')}</PanelLabel>
```
Comment (roxy-dao) [inline client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx:67]:
```suggestion
            <PanelLabel flex={0.5}>{t('heading.reference')}</PanelLabel>
```

Not sure why you were using 0.4? But anyways, it makes the 2 labels unalign when you change screen size 👀

File: client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx [hunk 2]
```diff
@@ -53,6 +63,31 @@ export const SidePanel = ({ data }: { data: RnRFormFragment }) => {
...
+                }
+                value={bufferedDetails?.theirReference ?? ''}
+                InputProps={{
+                  sx: { backgroundColor: 'white' },
+                }}
+              />
+            </PanelField>
+          </PanelRow>
+          <PanelRow>
+            <PanelLabel flex={0.4}>{t('heading.comment')}</PanelLabel>
```
Comment (roxy-dao) [inline client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx:82]:
```suggestion
            <PanelLabel flex={0.5}>{t('heading.comment')}</PanelLabel>
```

File: client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx [hunk 3]
```diff
@@ -53,6 +63,31 @@ export const SidePanel = ({ data }: { data: RnRFormFragment }) => {
             <PanelLabel>{t('label.created')}</PanelLabel>
             <PanelField>{localisedDate(data.createdDatetime)}</PanelField>
           </PanelRow>
+          <PanelRow>
```
Comment (roxy-dao) [inline client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx:66]: Can this be wrapped in its own grid and given more padding so it's not so close to the uneditable text... also do you think the two text boxes are too close to each other???

**Review by roxy-dao — CHANGES_REQUESTED** [review]
Lookin' good~

Just a few nits below 😭

File: client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx (hunk 1 above)
Comment (lache-melvin) [reply client/packages/requisitions/src/RnRForms/DetailView/SidePanel.tsx:67]: lol no idea, probably typo somehow 😁

**Comment by lache-melvin** [comment]
This better @roxy-dao ?

Screenshot 2024-10-17 at 12 45 09 PM

**Comment by lache-melvin** [comment]
Lol closed instead of commenting hehe sorry 🙈

**Comment by lache-melvin** [comment]
TWICE

**Review by roxy-dao — APPROVED** [review]
ngā mihi <3

**Pull request merged**